File: src/main/java/fr/epyi/metropiamod/item/ClothItem.java
package fr.epyi.metropiamod.item;

import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.nbt.CompoundNBT;

public class ClothItem extends Item {
    public ClothItem(Properties properties) {
        super(properties);
    }

    public static void setClothUrl(ItemStack stack, String url) {
        CompoundNBT compound = stack.getOrCreateTag();
        compound.putString("clothUrl", url);
    }

    public static String getClothUrl(ItemStack stack) {
        CompoundNBT compound = stack.getTag();
        if (compound != null && compound.contains("clothUrl")) {
            return compound.getString("clothUrl");
        }
        return "";
    }

    public static void setClothDirtiness(ItemStack stack, int dirtiness) {
        CompoundNBT compound = stack.getOrCreateTag();
        compound.putInt("clothDirtiness", dirtiness);
    }

    public static int getClothDirtiness(ItemStack stack) {
        CompoundNBT compound = stack.getTag();
        if (compound != null && compound.contains("clothDirtiness")) {
            return compound.getInt("clothDirtiness");
        }
        return 0;
    }

    public static void setClothArmor(ItemStack stack, int armor) {
        CompoundNBT compound = stack.getOrCreateTag();
        compound.putInt("clothArmor", armor);
    }

    public static int getClothArmor(ItemStack stack) {
        CompoundNBT compound = stack.getTag();
        if (compound != null && compound.contains("clothArmor")) {
            return compound.getInt("clothArmor");
        }
        return 0;
    }
}

File: src/main/java/fr/epyi/metropiamod/item/ClothItem.kt
package fr.epyi.metropiamod.item

import net.minecraft.item.Item
import net.minecraft.item.ItemStack

class ClothItem(properties: Properties?) : Item(properties) {
    companion object {
        fun setClothUrl(stack: ItemStack, url: String?) {
            val compound = stack.getOrCreateTag()
            compound.putString("clothUrl", url)
        }

        fun getClothUrl(stack: ItemStack): String {
            val compound = stack.tag
            if (compound != null && compound.contains("clothUrl")) {
                return compound.getString("clothUrl")
            }
            return ""
        }

        fun setClothDirtiness(stack: ItemStack, dirtiness: Int) {
            val compound = stack.getOrCreateTag()
            compound.putInt("clothDirtiness", dirtiness)
        }

        fun getClothDirtiness(stack: ItemStack): Int {
            val compound = stack.tag
            if (compound != null && compound.contains("clothDirtiness")) {
                return compound.getInt("clothDirtiness")
            }
            return 0
        }
    }
}